CMakeLists.txt

```
@@ -324,6 +324,15 @@ if(WIN32 AND MINGW)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -static-libstdc++")
 endif()
 
+if(WIN32)
+  include(CheckSymbolExists)
+  list(APPEND CMAKE_REQUIRED_LIBRARIES  "ws2_32")
+  check_symbol_exists(inet_pton "ws2tcpip.h" WIN32_INET_PTON_FOUND)
+  if(WIN32_INET_PTON_FOUND)
+    add_definitions(-DWIN32_HAS_INET_PTON)
+  endif()
+endif()
```

I think it's better to reset `CMAKE_REQUIRED_LIBRARIES` after this check to avoid any side effects, for example, if we will want to perform some other `check_symbol_exists` checks later.
```suggestion
  endif()
  list(REMOVE_ITEM CMAKE_REQUIRED_LIBRARIES "ws2_32")
endif()
```